packages/next/src/build/webpack/plugins/middleware-plugin.ts

```diff
@@ -249,6 +253,16 @@ function isDynamicCodeEvaluationAllowed(
   middlewareConfig?: MiddlewareConfig,
   rootDir?: string
 ) {
+  // Some packages are known to use `eval` but are safe to use in the Edge
+  // Runtime because the dynamic code will never be executed.
+  if (
+    KNOWN_SAFE_DYNAMIC_PACKAGES.some((pkg) =>
+      fileName.includes(`/node_modules/${pkg}/`.replace(/\//g, path.sep))
```

Comment:
this seems like it should use require.resolve instead of assuming it'll be at the root of node_modules

Reply:
Is there a specific case that this will happens? I think yarn/npm/pnpm all have it under `node_modules` and we're doing the same in other places too: https://github.com/vercel/next.js/blob/946c9c5c46091604b8a9c73260bfe26ece70de23/packages/next/src/build/webpack-config.ts#L1494-L1500

Reply:
Not directly under, necessarily, it could be heavily nested - and certainly not in yarn’s pnp mode.

Reply:
It’ll only be at the top level when there’s no version conflict. The entIre design of npm means that you can *never* safely hardcode node_modules anywhere, ever, and if it’s ever done, the tool is fundamentally broken. I urge you to use the actually correct means of locating a module directory.